### PR TITLE
Add Neo4jDataFrame#createNodes method and renameMap to mergeEdgeList

### PR DESCRIPTION
**Pull request by MrTomerLevi**
In this pull request, we're adding new capability to mergeEdgeList, creating a relationship between nodes having the same properties. For example:
(keanu:Person {name: 'Keanu'})-[:ACTED_WITH]->(Laurence:Person {name: "Laurence"})

In addition, we're adding new method createNodes to Neo4jDataFrame

### CODE REVIEW

**Comment by MrTomerLevi** [comment]
Can anyone help me understand why it fails on Travis?
On my end: mvn clean package or mvn test works fine.

**Comment by jexp** [comment]
Thanks a lot for your PR. I check it and let you know.

File: README.md [hunk 1]
```diff
@@ -255,11 +255,21 @@ There are a few different RDD's all named `Neo4jXxxRDD`
...
-  * property names from the sequence are used as column names for the data-frame, currently there is no name translation
-  * the result are sent in batches of 10000 to the graph 
+* Neo4jDataFrame provides:
+ 1.`mergeEdgeList(sc: SparkContext, dataFrame: DataFrame, source: (label,Seq[prop]), relationship: (type,Seq[prop]), target: (label,Seq[prop]))` to merge a DataFrame back into a Neo4j graph
+    * both nodes are merged by first property in sequence, all the others will be set on the entity
+    * relationships are merged between the two nodes and all properties from sequence will be set on the relationship
+    * property names from the sequence are used as column names for the data-frame, currently there is no name translation
+    * the result are sent in batches of 10000 to the graph
+ 2. `mergeEdgeList(sc: SparkContext, dataFrame: DataFrame, source: (String,Seq[String]), relationship: (String,Seq[String]), target: (String,Seq[String]), renameMap: Map[String,String])` to do the
+ same as the first method except for `renameMap` parameter.<br/>
```
Comment (jexp) [inline README.md]: can't we just add this one parameter to the first one with a default value (of an empty map).

Just curious why wouldn't you do that on the DF that gets passed in?

File: src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala [hunk 2]
```diff
@@ -16,25 +16,51 @@ import scala.collection.JavaConverters._
 object Neo4jDataFrame {
 
   def mergeEdgeList(sc: SparkContext, dataFrame: DataFrame, source: (String,Seq[String]), relationship: (String,Seq[String]), target: (String,Seq[String])): Unit = {
+    mergeEdgeList(sc, dataFrame, source,relationship, target, Map.empty)
+  }
+
+  def mergeEdgeList(sc: SparkContext, dataFrame: DataFrame, source: (String,Seq[String]), relationship: (String,Seq[String]), target: (String,Seq[String]), renameMap: Map[String,String]): Unit = {
+    val sourceLbel = renameMap.getOrElse(source._2.head, source._2.head)
```
Comment (jexp) [inline src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala]: typo Label

File: src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala [hunk 3]
```diff
@@ -16,25 +16,51 @@ import scala.collection.JavaConverters._
...
       MERGE (source)-[rel:`${relationship._1}`]->(target) ON CREATE SET rel += row.relationship
       """
     val partitions = Math.max(1,(dataFrame.count() / 10000).asInstanceOf[Int])
     val config = Neo4jConfig(sc.getConf)
     dataFrame.repartition(partitions).foreachPartition( rows => {
       val params: AnyRef = rows.map(r =>
         Map(
-          "source" -> source._2.map( c => (c, r.getAs[AnyRef](c))).toMap.asJava,
-          "target" -> target._2.map( c => (c, r.getAs[AnyRef](c))).toMap.asJava,
+          "source" -> source._2.map( c => (renameMap.getOrElse(c,c), r.getAs[AnyRef](c))).toMap.asJava,
```
Comment (jexp) [inline src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala]: if this turns out to be a perf issue, we could have two lists one for source and one for target names and build it up only once.

File: src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala [hunk 4]
```diff
@@ -16,25 +16,51 @@ import scala.collection.JavaConverters._
...
           .asJava).asJava
           execute(config, mergeStatement, Map("rows" -> params).asJava, write = true)
     })
   }
 
+
+  def createNodes(sc: SparkContext, dataFrame: DataFrame, nodes: (String,Seq[String])): Unit = {
+
+    val nodeLabel = nodes._2.head
+    val createStatement = s"""
```
Comment (jexp) [inline src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala]: no optional renaming here?

File: src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala [hunk 5]
```diff
@@ -16,25 +16,51 @@ import scala.collection.JavaConverters._
 object Neo4jDataFrame {
 
   def mergeEdgeList(sc: SparkContext, dataFrame: DataFrame, source: (String,Seq[String]), relationship: (String,Seq[String]), target: (String,Seq[String])): Unit = {
```
Comment (jexp) [inline src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala]: default value for renameMap perhaps better call it "renamedColumns"

**Comment by jexp** [comment]
Looks good, thanks a lot. Can you address the mini-comments?
Sorry I didn't see the PR when I built the release, but happy to build another release after this is merged.

**Comment by MrTomerLevi** [comment]
thanks, @jexp for the attention. Will take care of it

File: src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala (hunk 2 above)
Comment (MrTomerLevi) [reply src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala]: fixed

File: src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala (hunk 4 above)
Comment (MrTomerLevi) [reply src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala]: With nodes creation it's different I guess.
2 columns with the same name aren't possible.

File: src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala (hunk 5 above)
Comment (MrTomerLevi) [reply src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala]: done

File: src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala (hunk 3 above)
Comment (MrTomerLevi) [reply src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala]: We're getting values out of `renamedColumns` which is a `Map`.
It's an `O[1]` operation, isn't it?

File: README.md (hunk 1 above)
Comment (MrTomerLevi) [reply README.md]: @jexp thanks for reviewing this PR.

Agree, I removed the 2nd method and added a `renamedColumns` parameter to the original method (giving it a default value).

We can't do that in the DF that gets passed is as DF columns must have a unique name.
For example, if want to create ```(a:City)-[:HAS_ROAD_TO]->(b:City)``` relationship we need 2 columns named City

**Comment by MrTomerLevi** [comment]
Hi, @jexp could you please check out my last commit changes?

File: src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala (hunk 4 above)
Comment (jexp) [reply src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala]: I thought it as an additional aliasing.

Comment (jexp) [reply src/main/scala/org/neo4j/spark/Neo4jDataFrame.scala]: So it would be cool to add.

**Comment by jexp** [comment]
@MrTomerLevi thanks a lot. 

I think allowing the renaming also for nodes in general would be cool. As it saves an additional dataframe operation. Would you mind adding that?

**Comment by MrTomerLevi** [comment]
Hi @jexp, done.
Please review

**Comment by jexp** [comment]
Cool, thanks a lot!

**Pull request merged**